### ISSUE TITLE
Add "Clear Label" functionality for terrain labels

### ISSUE THREAD

**Issue by ecomitz** [wesnoth/wesnoth#8042]
### Describe the desired feature

Hey folks, yesterday (11/12/23) I was working on issue #7768 which describes a situation where if you erase all of the text on a label, the label, as we expect, disappears, but it does not for the other players. 

As I was thinking, reading, analyzing, et cetera, I saw how there was an option implemented to clear all labels in a multiplayer game, but not one to simply clear a single label from the game. 

To make a very long story short, my current thought is that it might be easier, and more useful, to simply implement an option where we can implement a remove label function, for its own merit, and simply call the remove label function to be called for all players in the game when one player erases the text in a terrain label. A kill two birds with one stone type of deal. 

The folks here were interested in it, too: https://forums.wesnoth.org/viewtopic.php?p=422613 (Although that was thirteen years ago!)

I can get started on it whenever, and I already have some ideas floating in the brain that need more refining and hammering - but first I wanted to ask about two things: 

1. Regardless of its bug fix potential as I described earlier, is this feature just something you folks would be interested in in general?

2. I understand currently we're in a "string freeze" - so if say, I, or anyone else, were to work on this, would it be better to save it for until after the string freeze?

**Comment by Wedge009** [comment]
Good on you for taking initiative. If you have some work you're willing to share, go ahead and make a pull request.

As far as I understand, the string freeze relates to strings marked for translation. If you're not introducing or modifying any of those, no worries. If you are, I say go ahead and submit your change anyway - the reviewers can decide on the timing of the merge.

**Comment by ecomitz** [comment]
> Good on you for taking initiative. If you have some work you're willing to share, go ahead and make a pull request.
> 
> As far as I understand, the string freeze relates to strings marked for translation. If you're not introducing or modifying any of those, no worries. If you are, I say go ahead and submit your change anyway - the reviewers can decide on the timing of any merge.

Good idea! If nothing else, perhaps, too, the code or the psuedocode for the button can be in the game, but not released until later, as well. Thanks for the suggestion!

**Comment by soliton-** [comment]
You can implement the function first without any GUI just to fix that bug. That can certainly go in for 1.18.

A GUI option sounds good as well but might not go into 1.18.
